src/backend/joanie/tests/core/test_flows_order.py

```diff
@@ -268,6 +268,131 @@ def test_flows_order_complete_failure_when_not_pending(self):
+    @override_settings(
+        JOANIE_LMS_BACKENDS=[
+            {
+                "API_TOKEN": "a_secure_api_token",
+                "BACKEND": "joanie.lms_handler.backends.openedx.OpenEdXLMSBackend",
+                "BASE_URL": "http://openedx.test",
+                "COURSE_REGEX": r"^.*/courses/(?P<course_id>.*)/course/?$",
+                "SELECTOR_REGEX": r".*",
+            }
+        ]
+    )
+    def test_flows_order_validate_auto_enroll(self):
+        """
+        When an order is validated, if one target course contains only one course run
+        and this one is opened for enrollment, the user should be automatically enrolled
+        """
+        course = factories.CourseFactory()
+        resource_link = (
+            "http://openedx.test/courses/course-v1:edx+000001+Demo_Course/course"
+        )
+        factories.CourseRunFactory(
+            course=course,
+            resource_link=resource_link,
+            state=CourseState.ONGOING_OPEN,
+            is_listed=False,
+        )
+        product = factories.ProductFactory(target_courses=[course], price="0.00")
+
+        url = "http://openedx.test/api/enrollment/v1/enrollment"
+        responses.add(
+            responses.POST,
+            url,
+            status=HTTPStatus.OK,
+            json={"is_active": True},
+        )
+
+        # Create an order
+        order = factories.OrderFactory(product=product)
+        order.flow.init()
+
+        self.assertEqual(order.state, enums.ORDER_STATE_COMPLETED)
+
+        self.assertEqual(len(responses.calls), 1)
+        self.assertEqual(responses.calls[0].request.url, url)
+        self.assertEqual(
+            responses.calls[0].request.headers["X-Edx-Api-Key"], "a_secure_api_token"
+        )
+        enrollment = Enrollment.objects.get(
+            user=order.owner, course_run__resource_link=resource_link
+        )
+        self.assertEqual(enrollment.state, "set")
+        self.assertEqual(
+            json.loads(responses.calls[0].request.body),
+            {
+                "is_active": enrollment.is_active,
+                "mode": "verified",
+                "user": enrollment.user.username,
+                "course_details": {"course_id": "course-v1:edx+000001+Demo_Course"},
+            },
+        )
+
+    @responses.activate
+    @override_settings(
+        JOANIE_LMS_BACKENDS=[
+            {
+                "API_TOKEN": "a_secure_api_token",
+                "BACKEND": "joanie.lms_handler.backends.openedx.OpenEdXLMSBackend",
+                "BASE_URL": "http://openedx.test",
+                "COURSE_REGEX": r"^.*/courses/(?P<course_id>.*)/course/?$",
+                "SELECTOR_REGEX": r".*",
+            }
+        ]
+    )
+    def test_flows_order_validate_auto_enroll_failure(self):
+        """
+        When an order is validated, if one target course contains only one course run
+        and this one is opened for enrollment, the user should be automatically enrolled
```

Comment:
```suggestion
        and this one is opened for enrollment, the user should be automatically enrolled.
```

Reply:
This is not the right PR to do those edits, I think.